state/machine.go

```diff
@@ -2241,6 +2227,20 @@ func (m *Machine) SetUpgradeCharmProfile(appName, chURL string) error {
 		if life == Dead || life == Dying {
 			return nil, ErrDead
 		}
+		charmURL, err := charm.ParseURL(chURL)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		ch, err := m.st.Charm(charmURL)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return nil, errors.Trace(err)
+			}
+		}
+		var emptyProfile bool
+		if ch == nil || (ch.LXDProfile() == nil || ch.LXDProfile().Empty()) {
+			emptyProfile = true
+		}
```

Comment:
This makes a lot of sense, I did have it this way a couple of times when developing, not entirely sure why I moved it...